### DISCUSSION TITLE
Does ShapeCrawler support shrink text on overflow in paragraphs?

### DISCUSSION THREAD

**Discussion by jamesmcroft** [ShapeCrawler/ShapeCrawler#163] [Q&A] [answered]
As the title suggest, wondering if ShapeCrawler supports the ability to insert text into a text box that has auto shrink enabled? Or potentially enabling auto shrink on the text box where appropriate.

**Comment by AdamShakhabov** [comment, marked as answer]
No, this functionality has not been implemented yet.

Will create issue for this.

**Comment by AdamShakhabov** [comment]
Support for "Shrink text on overflow" on _TextBox_ level is added in [**v0.26.0**](https://github.com/ShapeCrawler/ShapeCrawler/releases/tag/v0.26.0).

**Comment by valentimmx** [comment]
Hi there, how to use this feature? 
I'm setting up my Shape to "Shrink text on overflow" on .pptx but not works.

**Reply by AdamShakhabov** [reply]
Hello @valentimmx.

With "Shrink text on overflow" it should reduce font size if text content goes beyond the shape rectangle.

Could you share your case (.pptx file)? I would check this.

**Reply by valentimmx** [reply]
@adamshakhabov thanks for your very fast support!

Last night i found what i was doing wrong. 

I was setting text directly on paragraph `IParagraph.Text`, and not in `IAutoShape.TextFrame`.

I changed this and now it works!

Here my steps:
1. Changed the shape property to "Shrink text on overflow"
![image](https://user-images.githubusercontent.com/17087193/200051592-459988a5-8f08-41be-b472-b1b9345770c7.png)

2. Setted `IAutoShape.TextFrame.Text` rather than `paragraph.Text`

`Before (Doesn't Work)`
``` 
paragraph.Text = paragraph.Text.Replace("{UNDERSTANDING}", !string.IsNullOrEmpty(proposal.RequirementAnalysis) ? Regex.Replace(proposal.RequirementAnalysis, @"<[^>]+>|&nbsp;", "").Trim() : "-");
```
`After (Works)`
```
// Here is necessary to use "autoShape.TextFrame.Text" rather than "paragraph.Text" to wrap text inside shape.
autoShape.TextFrame.Text = paragraph.Text.Replace("{UNDERSTANDING}", !string.IsNullOrEmpty(proposal.RequirementAnalysis) ? Regex.Replace(proposal.RequirementAnalysis, @"<[^>]+>|&nbsp;", "").Trim() : "-");
```

**Comment by bryancw7** [comment]
Hi, I'm facing an issue. I'm setting up my TextBox to "Shrink text on overflow", but the text font size is reduced even before it goes beyond the rectangle. Do I make any mistake?

**Reply by AdamShakhabov** [reply]
Hello @bryancw7,

Could you please share a test presentation file to be able to reproduce this issue.

**Reply by bryancw7** [reply]
Thankyou for your response!

[template.pptx](https://github.com/user-attachments/files/18037741/template.pptx)
The TextBox is in the "Title and Content" Layout in Slide Master.
And here are the sample code that I used
```
var presentation = new Presentation("template.pptx");

presentation.Slides[0].Shapes[0].Text = "SUMMARY";
presentation.Slides.AddEmptySlide(SlideLayoutType.Object);

for (int i = 0; i < 10;  i++)
{
    presentation.Slides[1].Shapes[1].TextBox.Text += "Hello world!";
    presentation.SaveAs("result.pptx");
}
```

**Reply by bryancw7** [reply]
Any update on this? Can you reproduce the problem? Or I am making some mistake? Thanks @ashahabov

**Reply by AdamShakhabov** [reply]
Yes, I was able to reproduce this issue. I'm working on fixing it.

**Reply by AdamShakhabov** [reply]
The returning font size in this case is invalid. It should be tried again after fixing this bug https://github.com/ShapeCrawler/ShapeCrawler/issues/905
